servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/MulticastPublisher.java

```diff
@@ -15,355 +15,560 @@
+            }
         }
-    }
 
-    void requestIndividualSubscriber(MulticastSubscriber<T> subscriber) {
-        // This is the cumulative amount requested from this Subscriber.
-        // If we compare this with the cumulative amount we have requested upstream, we can deduce the delta to request.
-        final long individualSourceRequested = subscriber.sourceRequested();
-        for (;;) {
-            final long sourceRequested = this.sourceRequested;
-            if (sourceRequested >= individualSourceRequested) {
-                break;
+        private static final class RequestEvent<T extends MulticastLeafSubscriber<?>> {
+            final T subscriber;
+            final long n;
+
+            private RequestEvent(final T subscriber, final long n) {
+                this.subscriber = subscriber;
+                // sanitizeRequestN to avoid additional state for invalid requestN.
+                this.n = n == 0 ? -1 : n;
             }
-            if (sourceRequestedUpdater.compareAndSet(this, sourceRequested, individualSourceRequested)) {
-                subscription.request(individualSourceRequested - sourceRequested);
-                break;
+        }
+
+        private static final class CancelEvent<T extends MulticastLeafSubscriber<?>> {
+            final T subscriber;
+            final boolean cancelUpstream;
+
+            private CancelEvent(final T subscriber, final boolean cancelUpstream) {
+                this.subscriber = subscriber;
+                this.cancelUpstream = cancelUpstream;
             }
         }
     }
 
-    void cancelIndividualSubscriber(int subscriberIndex) {
-        Subscriber<? super T> subscriber = subscribers.getAndSet(subscriberIndex, noopSubscriber());
-        if (subscriber != NOOP_SUBSCRIBER && notCancelledCountUpdater.decrementAndGet(this) == 0) {
-            subscription.cancel();
+    private static final class MulticastFixedSubscriber<T> extends MulticastLeafSubscriber<T> implements Node {
+        private final int index;
+        private final MulticastPublisher<T>.State root;
+        private final Subscriber<? super T> subscriber;
+        private final Subscriber<? super T> ctxSubscriber;
+        /**
+         * Protected by {@link MulticastRootSubscriber} lock.
+         */
+        private long priorityQueueValue;
+        /**
+         * Protected by {@link MulticastRootSubscriber} lock.
+         */
+        private int priorityQueueIndex = INDEX_NOT_IN_QUEUE;
+
+        MulticastFixedSubscriber(final MulticastPublisher<T>.State root,
+                                 final Subscriber<? super T> subscriber, final AsyncContextMap contextMap,
+                                 final AsyncContextProvider contextProvider, final int index) {
+            this.root = root;
+            this.index = index;
+            this.subscriber = requireNonNull(subscriber);
+            // TODO(scott): SignalOffloader requires Subscriber methods invoked in order and terminal
```

Comment:
duplicate of https://github.com/apple/servicetalk/pull/1472#discussion_r609185214

now that @bondolo has merged the Executor removal changes this can be removed. SignalOffloader removal is coming soon IIUC :)